create_primers.py

```diff
@@ -0,0 +1,261 @@
+    parser.add_argument('primerprefix', help="prefix name to be added to each primer")
+    parser.add_argument('outfile', help='name of primer output file')
+    parser.add_argument('--startprimerlength', type=int, help='starting primer length', default=37)
+    parser.add_argument('--maxprimertm', type=float, help="Upper temperature limit for primers.", default=61)
+    parser.add_argument('--minprimertm', type=float, help="Lower temperature limit for primers.", default=60)
+    parser.add_argument('--minlength', type=int, help='Minimum primer length', default=25)
+    parser.add_argument('--maxlength', type=int, help='Maximum primer length', default=51)
+    return parser
+
+
+def ReverseComplement(seq):
+    """Returns reverse complement of sequence. Preserves upper/lower case."""
+    d = {'A':'T', 'T':'A', 'C':'G', 'G':'C', 'a':'t', 't':'a', 'c':'g', 'g':'c', 'N':'N', 'n':'n', 'S':'S', 's':'s', 'K':'M', 'k':'m'}
+    rc = [d[nt] for nt in seq]
+    rc.reverse()
+    return ''.join(rc)
+
+
+def CreateMutForOligosVarLength(seq, mutations_csv, primerlength, prefix, maxprimertm, minprimertm, maxlength, minlength):
+    """Creates oligos to tile a gene and introduce specified amino acid mutations at each site.
+
+    *seq* : sequence of the gene. The gene itself should be upper case,
+    including the start and stop codons. Any flanking regions should be lower
+    case.
+    There must be >= (primerlength - 3) / 2.0 nucleotides before the first codon
+    mutagenized and after the last codon mutagenized.
+
+    *mutations_csv* : csv file containing a table with the mutations the primers
+    will make to seq. The table should have one column, 'site', with site
+    numbers, and one column, 'mutant', with the single letter amino acid mutant
+    to make at that site. Each site can have mulitple mutants, or none. The site
+    number should denote the number of the codon in the uppercase gene, with the
+    start codon being 1.
+
+    *primerlength* : length of primers. Must be an odd number, so that equal length
+    flanking on each side.
+
+    *prefix* : string prefix attached to primer names.
+
+    Tiles primers across the gene in the forward direction. A unique primer is
+    made for each mutation specified by mutations_csv at each site.
+    Primers are named as follows:
+
+    f"{prefix}-for-mut{i}{mutant}" for 5' tiling primers, where i is the site
+    mutagenized in the gene and mutant is the amino acid the site is mutated to
+    using that primer.
+
+    Returns a list of all these primers as *(name, sequence)* 2-tuples.
+    """
+    n = len(seq)
+    assert primerlength % 2 == 1, "primer length not odd"
+    initial_flanklength = (primerlength - 3) // 2
+    upperseq = ''.join([nt for nt in seq if nt.istitle()])
```

Comment:
I think this should be using `nt.isupper()`. Technically `istitle()` works, but you're just comparing one nt at a time, so `isupper()` seems better.

You can also probably use `isupper()` on the seq somehow and not have to iterate to get all upper case letters back out, but I haven't confirmed the syntax for that.